src/dndbeyond/base/utils.js

```diff
@@ -465,6 +477,22 @@ async function sendRoll(character, rollType, fallback, args) {
             const operator = ["+", "-"].includes(modifier[0]) ? "" : "+"
             req.character.settings["custom-roll-dice"] = (req.character.settings["custom-roll-dice"] || "") + ` ${operator}${modifier}`;
         }
+
+        if(req.character.settings["effects-bless"] && req["to-hit"] && (req.type === "attack" || req.type === "spell-attack")) {
+            req.character.settings["custom-roll-dice"] = (req.character.settings["custom-roll-dice"] || "") + " +1d4";
+            addEffect(req, "Bless");
+        } else if(req.character.settings["effects-bane"] && req["to-hit"] && (req.type === "attack" || req.type === "spell-attack")) {
+            req.character.settings["custom-roll-dice"] = (req.character.settings["custom-roll-dice"] || "") + " -1d4";
+            addEffect(req, "Bane");
+        }
+
+        if(req.character.settings["effects-bless"] && req.type === "saving-throw") {
+            req["modifier"] += "+1d4";
+            addEffect(req, "Bless");
+        } else if(req.character.settings["effects-bane"] && req.type === "saving-throw") {
+            req["modifier"] += "-1d4";
+            addEffect(req, "Bane");
+        }
```

Comment:
The modifier is usually the +x modifier of the ability, so I'd say the bless/bane effect should still be added as a `custom-roll-dice` so you can bundle the check for saving-throw in the previous `if` statement.
```ks
if (req.character.settings["effects-bless"] && 
   (req.type === "saving-throw" || (["attack", "spell-attack"].includes(req.type) && req["to-hit"])) {
```
Actually, I think in the case of attack/spell-attack the custom-roll-dice isn't used if there is no to-hit, so you shouldn't need to check for it, so it could be simplified into
```js
if (req.character.settings["effects-bless"] && ["attack", "spell-attack", "saving-throw"].includes(req.type)) {
```